frontend/@posthog/ee/types.ts

```diff
@@ -0,0 +1,5 @@
+// NOTE: All exported items from the EE module _must_ be optionally defined to ensure we work well with FOSS
+export type PostHogEE = {
```

Comment:
this is really neat though! one or more types in here that define expectation and then a FOSS and EE licensed version.

Is it just typescript magic that ensure the EE version overwrites the FOSS version? 

I.e. imagine the case where we want code present in both versions but different behavoiur? Which might simply be something we explicitly say we don't support (or don't support this way...)

Reply:
We will cross that bridge when we come to it tbh. 
Typescript will import in order of the files mentioned which is how it works. So if the EE folder isn't present it will fallback to the base folder.